packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -3154,18 +3165,18 @@ export class StripeHelper {
     const subscription = await this.stripe.subscriptions.retrieve(
       (event.data.object as Stripe.Subscription).id
     );
-    // For PayPal customers, Stripe fails to send a `customer.updated` webhook
-    // when it automatically updates the `customer.currency` on subscription
-    // creation. Update the cached customer here to ensure `customer.currency`
-    // is not `null`.
-    // Since we can't easily differentiate a PayPal customer from others with
-    // an unexpanded subscription object, we update all customers in Firestore.
-    if (event.type === 'customer.subscription.created') {
+    // Update the customer if our copy of the customer is missing the currency.
+    // This could occur in some edge cases where the subscription is created
+    // before a user paying with paypal has the paypal agreement ID set on the
+    // user.
+    const customer = await this.expandResource(
+      subscription.customer,
+      CUSTOMER_RESOURCE
+    );
```

Comment:
The reason why I didn't previously fetch the customer from Stripe is that I know `fetchAndInsertCustomer` does that fetch already.

With the previous solution, we'd update our cached customer/subscriptions for all customers (Stripe and PayPal) on every `customer.subscription.updated` webhook but only make one Stripe call.

With this solution, we only update the cache in the applicable case, but we would also make a second Stripe call in the applicable case.

I think your solution is cleaner and more clear why we're doing this; just thinking about the trade-offs.

Reply:
In the happy path (the user is already in Stripe, with currency), there will be no Stripe calls on `expandResource` here.